### PR TITLE
[feat] remove registeration logic and adding language evaluation tasks.

### PR DESCRIPTION
**Pull request by Luodian**
* feat: Add language tasks and rebase to latest features of lm-evaluation-harness. The first batch support includes: mmlu, mmlu-pro, gsm8k, ifeval.

* refactor: Update with lm-evaluation-harness new features, including a specific class for Task/TaskOutput and use TaskManager to better handling the processing of tasks, group tasks and task alias, and tags. Tags will be future main feature and usage to specify if one task speficially belong to which group (use `tag` to denote the group).

* refactor: Update lmms_eval task/model registration logic for group tasks. Now we only need to register necessary tasks and models to speedup processing speed (up to 80% improvement and avoid many stuck issues). 

* Update pyproject.toml, remove some metrics installation since now these metrics would only be needed in certain tasks, users can install them when in need.

Before you open a pull-request, please check if a similar issue already exists or has been closed before.

### CODE REVIEW

**Comment by Luodian** [comment]
image

**Comment by Luodian** [comment]
OV-0.5b, MMLU = 15.6%

image

File: docs/commands.md
Comment (kcz358) [inline docs/commands.md:1]: The commands of the srt model seem to be outdated.

File: docs/run_examples.md
Comment (kcz358) [inline docs/run_examples.md:1]: The command seems to be outdated here also for the srt. I think it is some merge issue

File: lmms_eval/models/__init__.py [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "llava_sglang": "LlavaSglang",
     "llava_vid": "LlavaVid",
     "longva": "LongVA",
-    "mantis": "Mantis",
```
Comment (kcz358) [inline lmms_eval/models/__init__.py]: mantis got deleted during merging

File: lmms_eval/models/llava_vid.py [hunk 2]
```diff
@@ -96,7 +108,6 @@ def __init__(
         self.mm_spatial_pool_mode = mm_spatial_pool_mode
         self.max_frames_num = int(max_frames_num)
         self.mm_resampler_location = mm_pooling_position
-        self.mm_newline_position = mm_newline_position
```
Comment (kcz358) [inline lmms_eval/models/llava_vid.py]: The mm_newline_position error was introduced back again during merging for llava vid

File: lmms_eval/models/srt_api.py
Comment (kcz358) [inline lmms_eval/models/srt_api.py:1]: This file was also outdated. Seems the outside commits were lost during merging

**Review by kcz358 — CHANGES_REQUESTED** [review]
I think the merging options might have some problems in this PR. I will try to revert and re-merge the PR. Lots of files were revert to older version

**Review by kcz358 — APPROVED** [review]

**Pull request merged**